Make GifHub play nice with Refined GitHub

paulmolluzzo opened the pull request:
This fixes #6.

The solution is pretty simple: I add a class to the toolbar and then override the CSS from the [Refined GitHub](https://github.com/sindresorhus/refined-github) extension so that only the GifHub button is shown. This shouldn't break the current version for anyone, nor "break" anything for the Refined GitHub users either.

![gifhub-in-refined-github](https://cloud.githubusercontent.com/assets/737065/14081054/a4ca4ad4-f4d6-11e5-9468-a21e655aeb7a.png)

paulmolluzzo committed:
Make GifHub play nice with Refined GitHub

paulmolluzzo commented:
This doesn't work, sorry. I think I checked without properly refreshing. This is going to hide the toolbar just like in Refined GitHub for everyone, even if they don't have the other extension in their Chrome.

paulmolluzzo commented:
OK, I fixed it. And I restyled the buttons. If you want me to squash and push back up I can.

DrewML commented:
Thanks for contributing, @paulmolluzzo. The icon looks much better!

Quick Notes:
1. Can we add line endings to `extension.css` and `btn-template.html`?
2. This doesn't seem to work with partial page navigations. I believe the quick solution to that would be to pass `addClassToToolbar` to `onPartialRender`, which will re-invoke the callback each time pjax is used instead of a full page refresh.

paulmolluzzo commented:
> Can we add line endings to extension.css and btn-template.html?

Can you clarify this for me please? I don't know what you mean.

DrewML commented:
Ah sorry, just add a blank line to the bottom of those 2 files. It's a super small personal thing, but the unix `cat` command bleeds into your shell prompt when a file isn't terminated with a newline.

paulmolluzzo commented:
Oh, sure thing. You may want to put an editorconfig file in the repo to maintain this kind of style standard.

DrewML commented:
Good call - added :)

paulmolluzzo commented:
All should be fixed now.

DrewML commented on src/styles/extension.css:
```diff
@@ -88,4 +88,21 @@
+    display: block !important;
+}
+.gifhub-button .toolbar-group {
+    display: none;
```
This style has the side-effect of hiding the markdown toolbar for non `Refined GitHub` users on commit page comments

Screen-shot is with `Refined` disabled
![image](https://cloud.githubusercontent.com/assets/5233399/14096555/5c406196-f52f-11e5-95f7-74362ceed27f.png)

paulmolluzzo commented:
OK, I fixed the issue, and I have to admit it's a touch hacky. The reason I put the `setTimeout` in there is because I have to wait until the comment form opens to test if the markdown buttons are visible or not. I tried this with and without the RG extension running, and it's been working fine, but let me know if you still have issues.

paulmolluzzo commented:
Don't merge this yet please. There's a fix for Refined Github in the works here https://github.com/sindresorhus/refined-github/pull/118 and would make fixing this issue on GifHub a single line of CSS. :smile:

DrewML commented:
@paulmolluzzo Nice! Looking forward to it :)

paulmolluzzo commented:
OK, just waiting on that other one to get to `master` then I'll update this one in place. I'll make sure it's visible, keep the nicer icon I included in this version, and squash it all down to one commit.

paulmolluzzo commented:
@DrewML OK, I made this work with the `master` branch of the RG extension, and it looks nice. I squashed it all down to one commit and rebased on `master` of the GifHub repo.

Something strange is the empty line at the end of the extension.css and btn-template.html files. I have an empty line in both locally and there's no git diff pending, but the files _don't_ have the extra line on GitHub. IDK if there's something I borked along the way, or some config in ST or GitHub that I have on, but if that's the only objection then maybe merge and fix after? Or just pull down the PR I did and check that it's right locally? Super weird. ¯_(ツ)_/¯

DrewML commented:
Thank you for doing this @paulmolluzzo!

re: the line endings, looks good to me. GitHub won't actually show them, but they do show a little red icon when one is not present.

Just going to test this out real quick and then I'll merge in and push to the Chrome store.

DrewML commented:
Merged + published as 1.0.3.

paulmolluzzo commented:
![thevoice-the-voice-nbc-3ornjXIIShZ2MgyyHu](https://camo.githubusercontent.com/6ea47f5867f5d251a50cfb5708e4f200087b20e1/68747470733a2f2f6d65646961342e67697068792e636f6d2f6d656469612f336f726e6a58494953685a324d67797948752f67697068792e676966)
